src/integrationTest/java/com/hivemq/helmcharts/util/OperatorHelmChartUtils.java

```diff
@@ -0,0 +1,143 @@
+        return client;
+    }
+
+    public @NotNull ExecResult deployChart(File valuesPath,File operatorPath) throws IOException, InterruptedException {
+        //helm --kubeconfig /etc/rancher/k3s/k3s.yaml install hivemq /test/charts/hivemq-operator -f /test/src/test/resources/testValues.yaml
+        var outUpdate = container.execInContainer("/bin/helm",
+                "dependency",
+                "update",
+                operatorPath.getAbsolutePath()+"/");
+        assertTrue(outUpdate.getStderr().isEmpty());
+        return container.execInContainer("/bin/helm",
+                "--kubeconfig",
+                "/etc/rancher/k3s/k3s.yaml",
+                "install",
+                "hivemq",
+                operatorPath.getAbsolutePath(),
+                "-f",
+                valuesPath.getAbsolutePath());
+    }
+
+    public @NotNull String exposeService() throws IOException, ApiException, InterruptedException {
+        String kubeConfigYaml = container.getKubeConfigYaml();
+        var client = Config.fromConfig(new StringReader(kubeConfigYaml));
+        var api = new CoreV1Api(client);
+        var nodes = api.listNode(null, null, null, null, null, null, null, null, null, null);
+        assertTrue(nodes.getItems().size() > 0);
+        V1Pod foundPods = waitForHiveMQCluster(api);
+        assertNotNull(foundPods);
+        //When the pod is running the cluster needs time to be ready
+        var podLog = new PodLogs(client);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (var is = podLog.streamNamespacedPodLog(foundPods)) {
+            var scanner = new Scanner(is).useDelimiter("\n");
+            while (scanner.hasNext()) {
+                var s = scanner.next();
+                stringBuilder.append(s);
+                stringBuilder.append("\n");
+                if (s.contains("Started HiveMQ in")) {
+                    System.out.println("Broker Ready");
+                    break;
+                }
+            }
+        }
+        //ByteStreams.copy(is,System.out);
+        return stringBuilder.toString();
+    }
+
+    private @Nullable V1Pod waitForHiveMQCluster(@NotNull CoreV1Api api) throws ApiException, InterruptedException, IOException {
+        int retries = 20;
+        boolean serviceReady = false;
+        while (retries >= 0) {
+            List<V1Pod> defaultPods = api.listNamespacedPod("default",
+                    null, null, null, null,
+                    "app=hivemq", null, null, null,
+                    null, null).getItems();
+
+            if (defaultPods.size() == 1) {
+                // As soon as the pod is created we can expose the service
+                if (!serviceReady) {
+                    var outExpose = container.execInContainer("/bin/kubectl",
+                            "expose",
+                            "deployment",
+                            "hivemq",
+                            "--type=LoadBalancer",
```

Comment:
this should not be done here, you can use a patch like described in the values.yaml: 
https://github.com/hivemq/helm-charts/blob/master/charts/hivemq-operator/values.yaml#L245-L246
and then connect directly to the LoadBalancer without going to the raw resources so much